src/bundle.c

```diff
@@ -349,14 +349,17 @@ gboolean resign_bundle(const gchar *inpath, const gchar *outpath, GError **error
 	return res;
 }
 
-gboolean check_bundle(const gchar *bundlename, gsize *size, gboolean verify, GError **error) {
+gboolean check_bundle(const gchar *bundlename, RaucBundle **bundle, gboolean verify, GError **error) {
 	GError *ierror = NULL;
 	GBytes *sig = NULL;
 	GFile *bundlefile = NULL;
 	GFileInputStream *bundlestream = NULL;
 	guint64 sigsize;
 	goffset offset;
 	gboolean res = FALSE;
+	RaucBundle *ibundle = g_new0(RaucBundle, 1);
+
+	ibundle->path = g_strdup(bundlename);
 
```

Comment:
We should check that bundle points to a NULL pointer or is NULL itself. Otherwise we'd leak a RaucBundle later when passing the new RaucBundle.

Reply:
Added `g_return_val_if_fail()` check for this.